board/opendingux/package/blockattack/default.gcw0.desktop

```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Block Attack
+Comment=Switch blocks so they match
+Exec=env SDL_GAMECONTROLLERCONFIG=190000006c696e6b6465762064657600,RG350,platform:Linux,x:b3,a:b0,b:b1,y:b2,back:b8,start:b9,dpleft:h0.8,dpdown:h0.4,dpright:h0.2,dpup:h0.1,leftshoulder:b4,lefttrigger:b6,rightshoulder:b5,righttrigger:b7,leftstick:b10,rightstick:b11,leftx:a0,lefty:a1,rightx:a2,righty:a3, ./bin/blockattack
```

Comment:
`SDL_GAMECONTROLLERCONFIG` won't be necessary with the new firmwares, is only here for backwards compatibility with older firmwares, can be removed once most users upgrade

Reply:
@glebm does this actually work? I'm having some difficulty getting this controller config working.

Both in a custom game I've built, and using sdl2's `controllermap` test binary, I'm getting very strange results. Could you tell me how you came to this configuration?

It seems to me that R1 is backspace, SEL is escape, and X is space. I don't know if I'm getting something wrong, but my SDL2 controller code works great on desktop with xb1 and ps4 controllers using a char* of controller configs, but adding this one doesn't get my code working on the RG350.

Reply:
is this because my original firmware is using gpio-keys?
```
[    1.060000] input: gpio-keys as /devices/platform/gpio-keys/input/input3
[    1.060000] input input3: Dropping handle 'kbd' from handler 'kbd'
[    1.060000] platform linkdev: Connected to device gpio-keys
```

Reply:
> It seems to me that R1 is backspace, SEL is escape, and X is space

From your description it looks like you haven't specified `X-OD-NeedsJoystick=true` in the `.desktop` file.

Reply:
Awesome thanks, the `X-OD-NeedsJoystick=true` certainly helped, but I think I'm behind on the firmware.

Could you point me in the right direction for firmware updates? There's a lot of weird "custom firmware" threads on the dingoonity forum, but which is the most official?

This one?
https://rs97.bitgala.xyz/RG-350/Latest%20Firmware/

Reply:
See https://github.com/retrogamehandheld/RG-350/wiki/Firmware-and-Software-Updates
Most of the discussion happens on this Discord https://discord.gg/p4uRmCd

Reply:
Nice, it works! Thanks!